src/ui/AuthenticationProvider/AuthenticationProvider.ts

```diff
@@ -252,6 +252,12 @@ export class AuthenticationProvider extends Component {
   }
 
   private handleQueryError(args: IQueryErrorEventArgs) {
+    if (args.error.name === 'InvalidTokenException') {
+      localStorage.removeItem(accessTokenStorageKey);
+      this._window.location.reload();
```

Comment:
What exactly are we trying to trigger by reloading? Could we do it directly?

Reply:
A reload would reinitialize the page. Since there is no access token in localstorage, it would start the SAML authentication loop, redirecting a user to the third-party login page.

Starting the loop directly is possible, but would take more code changes because we would need to store the initial API key in memory, replace the invalid token with the api key, and then perform a request. I felt reloading the page was a simpler approach to solve the problem, especially since the user was going to be redirected.